pkg/scheduler/preemption/preemption.go

```diff
@@ -240,32 +240,50 @@ func resourcesRequiringPreemption(assignment flavorassigner.Assignment) resource
 // preempting workload needs.
 func findCandidates(wl *kueue.Workload, cq *cache.ClusterQueue, resPerFlv resourcesPerFlavor) []*workload.Info {
 	var candidates []*workload.Info
-	cqs := sets.New(cq)
-	if cq.Cohort != nil && cq.Preemption.ReclaimWithinCohort != kueue.PreemptionPolicyNever {
-		cqs = cq.Cohort.Members
+	wlPriority := priority.Priority(wl)
+
+	if cq.Preemption.WithinClusterQueue != kueue.PreemptionPolicyNever {
```

Comment:
What's the reason to swap the order to check `ReclaimWithinCohort` and `WithinClusterQueue`?

I'm not saying it would break something, but just want to understand if it's necessary. By swapping the order we get a more complex diff, so it's hard to tell if something changed.